sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/highlighting/XmlHighlightingTest.java

```diff
@@ -0,0 +1,237 @@
+    assertEquals(5, highlightingData.size());
+    assertData(highlightingData.get(2), 5, 14, "k");
+    assertData(highlightingData.get(3), 22, 25, "k");
+  }
+
+  @Test
+  public void testHighlightTag() throws XMLStreamException {
+    List<HighlightingData> highlightingData = new XMLHighlighting("<tr></tr>").getHighlightingData();
+    assertEquals(3, highlightingData.size());
+    // <tr
+    assertData(highlightingData.get(0), 0, 3, "k");
+    // >
+    assertData(highlightingData.get(1), 3, 4, "k");
+    // </tr>
+    assertData(highlightingData.get(2), 4, 9, "k");
+  }
+
+  @Test
+  public void testEmptyElement() throws Exception {
+    List<HighlightingData> highlightingData = new XMLHighlighting("<br/>").getHighlightingData();
+    assertEquals(3, highlightingData.size());
+    // <br
+    assertData(highlightingData.get(0), 0, 3, "k");
+    // /
+    assertData(highlightingData.get(2), 3, 4, "k");
+    // >
+    assertData(highlightingData.get(1), 4, 5, "k");
+  }
+
+  @Test
+  public void testSpacesInside() throws Exception {
+    List<HighlightingData> highlightingData = new XMLHighlighting("<tag > </tag >").getHighlightingData();
+    assertEquals(3, highlightingData.size());
+    assertData(highlightingData.get(0), 0, 4, "k");
+    assertData(highlightingData.get(1), 5, 6, "k");
+    assertData(highlightingData.get(2), 7, 14, "k");
+
+    highlightingData = new XMLHighlighting("<tag />").getHighlightingData();
+    assertEquals(3, highlightingData.size());
+    assertData(highlightingData.get(0), 0, 4, "k");
+    assertData(highlightingData.get(2), 5, 6, "k");
+    assertData(highlightingData.get(1), 6, 7, "k");
+  }
+
+  @Test
+  public void testHighlightTagWithNamespace() throws XMLStreamException {
+    List<HighlightingData> highlightingData = new XMLHighlighting("<tag xmlns:x='url'>\n<x:table>  </x:table></tag>").getHighlightingData();
+    assertEquals(8, highlightingData.size());
+    assertData(highlightingData.get(1), 5, 12, "c");
+    assertData(highlightingData.get(2), 13, 18, "s");
+    assertData(highlightingData.get(4), 20, 28, "k");
+    assertData(highlightingData.get(5), 28, 29, "k");
+  }
+
+  @Test
+  public void testXMLHeader() throws XMLStreamException {
+    List<HighlightingData> highlightingData = new XMLHighlighting("<?xml version=\"1.0\" encoding=\"UTF-8\" ?> <tag/>").getHighlightingData();
+    assertEquals(9, highlightingData.size());
+    // <?xml
+    assertData(highlightingData.get(0), 0, 5, "k");
+    // ?>
+    assertData(highlightingData.get(5), 37, 39, "k");
+
+    // version
+    assertData(highlightingData.get(1), 6, 13, "c");
+    // "1.0"
+    assertData(highlightingData.get(2), 14, 19, "s");
+    // encoding
+    assertData(highlightingData.get(3), 20, 28, "c");
+    // "UTF-8"
+    assertData(highlightingData.get(4), 29, 36, "s");
+  }
+
+  @Test
+  public void entity() throws Exception {
+    assertThat(new XMLHighlighting("<a>&ouml;</a>").getHighlightingData());
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Complete the assertion. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2970)